lib/ansible/module_utils/basic.py

```diff
@@ -1148,16 +1148,22 @@ def set_attributes_if_different(self, path, attributes, changed, diff=None, expa
             return True
 
         existing = self.get_file_attributes(b_path)
+        existing_flags = set(existing.get('attr_flags', ''))
+        delta_flags = set(attr for attr in attributes if attr in FILE_ATTRIBUTES)
+
+        if attributes.startswith('+'):
+            new_flags = existing_flags | delta_flags
+        elif attributes.startswith('-'):
+            new_flags = existing_flags - delta_flags
+        elif attributes.startswith('='):
```

Comment:
The `elif attributes.startswith('='):` isn't required. The `else` arm has the same code.